servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -0,0 +1,451 @@
+      // Failed to allocate a temp block, let Evictor kick in to ensure sufficient space available.
+
+      // Upgrade to write lock to guard evictor. This is not a real "upgrade" and things can
+      // happen between unlock and lock.
+      mEvictionLock.readLock().unlock();
+      mEvictionLock.writeLock().lock();
+      try {
+        freeSpaceInternal(userId, initialBlockSize, location);
+      } finally {
+        // Downgrade to read lock again after eviction. Lock before unlock is intentional.
+        mEvictionLock.readLock().lock();
+        mEvictionLock.writeLock().unlock();
+      }
+      tempBlock = mAllocator.allocateBlock(userId, blockId, initialBlockSize, location);
+      Preconditions.checkNotNull(tempBlock, "Cannot allocate block %s:", blockId);
+    }
+    // Add allocated temp block to metadata manager
+    mMetaManager.addTempBlockMeta(tempBlock);
+    return tempBlock;
+  }
+
+  // Commit a temp block. This method requires no eviction lock acquired.
+  private void commitBlockNoLock(long userId, long blockId, TempBlockMeta tempBlockMeta)
+      throws IOException {
+    if (mMetaManager.hasBlockMeta(blockId)) {
+      throw new IOException("Failed to commit block " + blockId + ": block is committed");
+    }
+    // Check the userId is the owner of this temp block
+    long ownerUserId = tempBlockMeta.getUserId();
+    if (ownerUserId != userId) {
+      throw new IOException("Failed to commit temp block " + blockId + ": ownerUserId "
+          + ownerUserId + " but userId " + userId);
+    }
+    String sourcePath = tempBlockMeta.getPath();
+    String destPath = tempBlockMeta.getCommitPath();
+    boolean renamed = new File(sourcePath).renameTo(new File(destPath));
+    if (!renamed) {
+      throw new IOException("Failed to commit temp block " + blockId + ": cannot rename from "
+          + sourcePath + " to " + destPath);
+    }
+    mMetaManager.commitTempBlockMeta(tempBlockMeta);
+  }
+
+  // Abort a temp block. This method requires no eviction lock acquired.
+  private void abortBlockNoLock(long userId, long blockId) throws IOException {
+    if (mMetaManager.hasBlockMeta(blockId)) {
+      throw new IOException("Failed to abort block " + blockId + ": block is committed");
+    }
+
+    TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
+    // Check the userId is the owner of this temp block
+    long ownerUserId = tempBlockMeta.getUserId();
+    if (ownerUserId != userId) {
+      throw new IOException("Failed to abort temp block " + blockId + ": ownerUserId "
+          + ownerUserId + " but userId " + userId);
+    }
+    String path = tempBlockMeta.getPath();
+    boolean deleted = new File(path).delete();
+    if (!deleted) {
+      throw new IOException("Failed to abort temp block " + blockId + ": cannot delete " + path);
+    }
+    mMetaManager.abortTempBlockMeta(tempBlockMeta);
+  }
+
+  // Move a block. This method requires block lock required but no eviction lock acquired.
```

Comment:
Both the blockLock and the evictionLock are required before calling this. The comments don't seem to reflect that.